Crypto/src/X509Certificate.cpp

```diff
@@ -255,7 +261,7 @@ void X509Certificate::extractNames(std::string& cmnName, std::set<std::string>&
 			const GENERAL_NAME* name = sk_GENERAL_NAME_value(names, i);
 			if (name->type == GEN_DNS)
 			{
-				const char* data = reinterpret_cast<char*>(ASN1_STRING_data(name->d.ia5));
+				const char* data = reinterpret_cast<char*>(ASN1_STRING_get0_data(name->d.ia5));
```

Comment:
My C++ is poor but note that ASN1_STRING_get0_data returns const unsigned char* whereas ASN1_STRING_data returns unsigned char*. There might be a const_cast needed.

Reply:
Thanks, will fix.